src/steadystateproblem.cpp

```diff
@@ -264,38 +263,30 @@ void SteadystateProblem::getSteadystateSimulation(ReturnData *rdata, Solver *sol
         /* increase counter, check for maxsteps */
         steps_newton++;
         if (steps_newton >= solver->getMaxSteps() && !converged)
-            throw NewtonFailure(AMICI_TOO_MUCH_WORK, "getSteadystateSimulation");
+            throw NewtonFailure(AMICI_TOO_MUCH_WORK, "exceeded maximum number of steps");
     }
+    rdata->newton_numsteps[0] = steps_newton;
     if (solver->getSensitivityOrder()>SensitivityOrder::none)
         solver->getSens(t, sx);
 }
 
 std::unique_ptr<CVodeSolver> SteadystateProblem::createSteadystateSimSolver(
         Solver *solver, Model *model, realtype tstart)
 {
-
-    
-    /* Create new CVode object */
+    /* Create new CVode solver object */
+    auto simsolver_ptr = dynamic_cast<CVodeSolver*>(solver->clone());
```

Comment:
Leaks `simsolver_ptr` on `throw` below. Put into `std::unique<Solver>` and convert later, or use as is